### ISSUE TITLE
Selenium Webdriver breaking rspec/gemfile

### ISSUE THREAD

**Issue by Triffanys101** [makersacademy/slack-overflow#61]
So everything was going well until I tried to incorporate likes into my instagram. For this I wrote the following test:

```feature 'Liking pictures' do
      it "users can like a picture, which then updates the 'like' count by one", js: true do
         click_link 'Like'
         expect(page).to have_content '1 like'
         expect(page).not_to have_content '0 likes'    
    end ```

This gave me the following error:

 ```Failure/Error: visit '/'
     LoadError:
       Capybara's selenium driver is unable to load `selenium-webdriver`, please install the gem and add `gem 'selenium-webdriver'` to your Gemfile if you are using bundler.```

So I installed 'selenium-webdriver' as it is a gem i've used before and I've been taught to follow the error messages when coding. Before installing this I had 29 passing tests and just the one failing test above. After installing selenium-webdriver I have 14 failing tests that are all saying:

```Failure/Error: click_link 'Upload a picture'
     Capybara::ElementNotFound:
       Unable to find link "Upload a picture"```

I therefore went to local host and the website is working and I can see the link, click on it etc. Therefore I assumed it was the selenium-webdriver gem messing up my tests so I uninstalled it, removed it from my gem file and removed the test expecting rspec to return to 29 green tests.

Unfortunately this was not the case and I still have 13 failing tests all referencing the above error above. I've tried returning to a previous commit from before I had even written the above test yet when I run rspec I still get 13 failing tests... I have also tried git cloning the repository onto a different computer but that couldn't even complete the bundle and therefore was unable to run rspec. I'm wondering if I have a bug in my gem file that I'm just not spotting?

This is my gemfile: 

```source 'https://rubygems.org'
gem 'devise'
gem 'paperclip'
gem 'rails', '4.2.3'
gem 'pg'
gem 'sass-rails', '~> 5.0'
gem 'uglifier', '>= 1.3.0'
gem 'coffee-rails', '~> 4.1.0'
gem 'jquery-rails'
gem 'turbolinks'
gem 'jbuilder', '~> 2.0'
gem 'sdoc', '~> 0.4.0', group: :doc```

```group :development, :test do
  gem 'byebug'
  gem 'web-console', '~> 2.0'
  gem 'spring'
end```

```group :test do
  gem 'rspec-rails'
  gem 'capybara'
  gem 'rspec-collection_matchers'
  gem 'shoulda'
end```

This is a link to my github repo any help would be appreciated: https://github.com/Triffanys101/instagram-challenge/tree/master/instagram_clone

**Comment by dwatson62** [comment]
I just cloned down your project, and the tests were passing for me. When you removed selenium webdriver did you do bundle update? That builds up your Gemfile.lock from scratch. If not, then selenium would've still been installed and breaking your tests.

For your problem though, its a database issue. I ran byebug before your tests and saw that there was a test user already in the database, so when you tried to sign up a new test user before each test it was failing because of duplicate emails. I gem installed and setup database cleaner as per the yelp clone instructions https://github.com/makersacademy/course/blob/master/rails/yelpv3.md#testing-js-on-web-pages so that it wipes the database clean, and all the tests are passing now.
